core/edr-core/src/main/java/org/eclipse/tractusx/edc/edr/core/manager/EdrManagerImpl.java

```diff
@@ -0,0 +1,368 @@
+        monitor.debug(format("Edr entry %s is now in state %s.", edrEntry.getId(), from(edrEntry.getState())));
+    }
+
+
+    private StateProcessorImpl<EndpointDataReferenceEntry> processEdrInState(EndpointDataReferenceEntryStates state, Function<EndpointDataReferenceEntry, Boolean> function) {
+        var filter = new Criterion[]{ hasState(state.code()) };
+        return new StateProcessorImpl<>(() -> edrCache.nextNotLeased(batchSize, filter), telemetry.contextPropagationMiddleware(function));
+    }
+
+    private ContractRequest createContractRequest(NegotiateEdrRequest request) {
+        var callbacks = Stream.concat(request.getCallbackAddresses().stream(), Stream.of(LOCAL_CALLBACK)).collect(Collectors.toList());
+
+        var requestData = ContractRequestData.Builder.newInstance()
+                .contractOffer(request.getOffer())
+                .protocol(request.getProtocol())
+                .counterPartyAddress(request.getConnectorAddress())
+                .connectorId(request.getConnectorId())
+                .build();
+
+        return ContractRequest.Builder.newInstance()
+                .requestData(requestData)
+                .callbackAddresses(callbacks).build();
+    }
+
+    private boolean processNegotiated(EndpointDataReferenceEntry edrEntry) {
+        if (isAboutToExpire(edrEntry)) {
+            return entityRetryProcessFactory.doSyncProcess(edrEntry, () -> fireTransferProcess(edrEntry))
+                    .onDelay(this::breakLease)
+                    .onSuccess((n, result) -> transitionToRefreshing(n))
+                    .onFailure((n, throwable) -> transitionToNegotiated(n))
+                    .onFatalError((n, failure) -> transitionToError(n, failure.getFailureDetail()))
+                    .onRetryExhausted((n, failure) -> transitionToError(n, format("Failed renew EDR token: %s", failure.getFailureDetail())))
+                    .execute("Start an EDR token renewal");
+        } else {
+            breakLease(edrEntry);
+            return false;
+        }
+    }
+
+    private boolean processExpired(EndpointDataReferenceEntry edrEntry) {
+        if (shouldBeRemoved(edrEntry)) {
+            return entityRetryProcessFactory.doSyncProcess(edrEntry, () -> deleteEntry(edrEntry))
+                    .onDelay(this::breakLease)
+                    .onSuccess((n, result) -> {
+                    })
+                    .onFailure((n, throwable) -> transitionToExpired(n))
+                    .onFatalError((n, failure) -> transitionToError(n, failure.getFailureDetail()))
+                    .onRetryExhausted((n, failure) -> transitionToError(n, format("Failed delete EDR token: %s", failure.getFailureDetail())))
+                    .execute("Start an EDR token deletion");
+        } else {
+            breakLease(edrEntry);
+            return false;
+        }
+    }
+
+    private StatusResult<Void> deleteEntry(EndpointDataReferenceEntry entry) {
+        this.transitionToDeleted(entry);
```

Comment:
Actually this is tricky :D, i cannot the delete directly the thing because it's under lease, so i needed i way to transition
 to a state that's not picked by the state machine, which is then safe to delete. Rethinking this probably i will do a custom delete method in the store interface for being able to "unsafe" delete  stuff